src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql

```diff
@@ -1336,6 +1321,82 @@ AS
     COMMIT TRANSACTION
 GO
 
+--
+-- STORED PROCEDURE
+--     Acquires export jobs.
+--
+-- DESCRIPTION
+--     Timestamps the available export jobs and sets their statuses to running.
+--
+-- PARAMETERS
+--     @expirationDateTime
+--         * The time before which an export job is considered stale
+--     @maximumNumberOfConcurrentJobsAllowed
+--         * The maximum number of running jobs we can have at once
+--     @heartbeatDateTime
+--         * The time to be stamped on the newly running export jobs
+--
+-- RETURN VALUE
+--     The updated jobs that are now running.
+--
+CREATE PROCEDURE dbo.AcquireExportJobs
+    @expirationDateTime datetimeoffset(7),
+    @maximumNumberOfConcurrentJobsAllowed int,
+    @heartbeatDateTime datetimeoffset(7)
+AS
+    SET NOCOUNT ON
+    SET XACT_ABORT ON
+
+    SET TRANSACTION ISOLATION LEVEL SERIALIZABLE
+    BEGIN TRANSACTION
+
+    -- Get the number of jobs that are running and not stale.
+    -- Acquire a table lock, which will be held for the entire transaction.
+    DECLARE @numberOfRunningJobs int
+    SELECT @numberOfRunningJobs = COUNT(*) FROM dbo.ExportJob WITH (TABLOCKX) WHERE Status = 'Running' AND HeartbeatDateTime > @expirationDateTime
+
+    -- Determine how many available jobs we can pick up.
+    DECLARE @limit int = @maximumNumberOfConcurrentJobsAllowed - @numberOfRunningJobs;
+
+    DECLARE @availableJobs TABLE (Id varchar(64) COLLATE Latin1_General_100_CS_AS NOT NULL, JobVersion binary(8) NOT NULL)
+
+    -- Get the available jobs, which are export jobs that are queued or stale.
+    INSERT INTO @availableJobs
+    SELECT TOP (@limit) Id, JobVersion
+    FROM dbo.ExportJob
+    WHERE (Status = 'Queued' OR (Status = 'Running' AND HeartbeatDateTime <= @expirationDateTime))
+    ORDER BY HeartbeatDateTime, QueuedDateTime
+
+    DECLARE @updatedJobs TABLE (RawJobRecord varchar(max) NOT NULL, JobVersion binary(8) NOT NULL)
```

Comment:
I think (but I'm not certain) that you can omit the table variable declaration, the `INTO` clause in the `UPDATE` below, and the subsequent `SELECT`. The `OUTPUT` clause should be included in the stored procedure's resultset because of the the `OUTPUT` clause.

Reply:
Yes you're right! Awesome. Removed in 7caed89.